### PR TITLE
⚠ Recover panics per default

### LINKED ISSUE #2167: RecoverPanic - true by default
Just curious why this isn't set to true by default. Would anyone object if I made a PR to enable by default? https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/config/controller.go#L44

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

Recover panics in reconcilers & webhooks per default. Also adds corresponding metrics.

Fixes #2167

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold

I'll do a bit of testing

File: pkg/builder/webhook.go [hunk 1]
```diff
@@ -84,8 +84,9 @@ func (blder *WebhookBuilder) WithLogConstructor(logConstructor func(base logr.Lo
 }
 
 // RecoverPanic indicates whether panics caused by the webhook should be recovered.
-func (blder *WebhookBuilder) RecoverPanic() *WebhookBuilder {
-	blder.recoverPanic = true
+// Defaults to true.
+func (blder *WebhookBuilder) RecoverPanic(recoverPanic *bool) *WebhookBuilder {
```
Comment (alvaroaleman) [inline pkg/builder/webhook.go:88]: Why take a bool pointer and not a value?

Comment (sbueringer) [reply pkg/builder/webhook.go:88]: Also sort of tried to be consistent with the  RecoverPanic *bool option we have for controllres, but I'm aware that we don't need that for these funcs :)

Comment (sbueringer) [reply pkg/builder/webhook.go]: Fixed

**Comment by sbueringer** [comment]
@alvaroaleman fixed, PTAL :)

**Pull request merged**